src/utils/url.js

```diff
@@ -20,6 +20,8 @@ export const REQUEST_DESCRIPTOR_VALUES_SEPARATOR = '!';
 export const TRAILING_SLASH_RE                   = /\/$/;
 export const SPECIAL_PAGES                       = ['about:blank', 'about:error'];
 
+export const DEFAULT_PORT = ':80';
```

Comment:
Don't duplicate constants.
Create a shared function `omitDefaultPort ()` and use it on client and server.
Use implementation from server side - https://github.com/DevExpress/testcafe-hammerhead/blob/4ea7ec3b78993f48db4863eee8852391dc12f46e/src/request-pipeline/context.js#L82.

Reply:
Add tests for `http` and `https` protocol values.